### PR TITLE
'Musical symbols scale' option

### PR DESCRIPTION
**Pull request by XiaoMigros**
Rebase of #20772
Resolves: #19366
Resolves: #17885

This PR changes the following:
- An option for tuplets to use SMuFL number symbols, rather than plain text
- Pedal line text now uses the musical _symbols_ font instead of musical _text_ font.
-  A new text style property, 'Musical symbols scale', to control the size of musical symbols in text-based elements (similar to the existing control for dynamics). This option is available for tuplets, ottava, and pedal lines.

### CODE REVIEW

**Comment by rettinghaus** [comment]
Saving 

`<text>&lt;sym&gt;tuplet3&lt;/sym&gt;</text>`

instead of 

`<text>3</text>`

doesn't seem to be a good idea.

**Comment by XiaoMigros** [comment]
The utests agree with you! A more elegant solution is underway...

**Review by oktophonie — APPROVED** [review]

**Review by mike-spa — APPROVED** [review]

**Review by bkunda — APPROVED** [review]

File: src/notation/view/widgets/editstyle.cpp [hunk 1]
```diff
@@ -2588,6 +2605,13 @@ void EditStyle::textStyleChanged(int row)
     styleName->setEnabled(int(tid) >= int(TextStyleType::USER1));
     resetTextStyleName->setEnabled(styleName->text() != TConv::translatedUserName(tid));
 
+    tupletUseSymbols->setVisible(int(tid) == int(TextStyleType::TUPLET));
+    resetTupletUseSymbols->setVisible(int(tid) == int(TextStyleType::TUPLET));
+    row_textStyleMusicalSymbolsScale->setVisible(int(tid) == int(TextStyleType::TUPLET)
+                                                 || int(tid) == int(TextStyleType::OTTAVA) || int(tid) == int(TextStyleType::PEDAL)
+                                                 || int(tid) == int(TextStyleType::HARP_PEDAL_DIAGRAM));
+    row_textStyleMusicalSymbolsScale->setEnabled(int(tid) != int(TextStyleType::TUPLET) || tupletUseSymbols->isChecked());
```
Comment (cbjeukendrup) [inline src/notation/view/widgets/editstyle.cpp:2613]: Seems that the `int()` conversions are not necessary here (the reason they are used above is probably the `>=` operator)

File: src/notation/view/widgets/editstyle.cpp [hunk 2]
```diff
@@ -2602,7 +2626,11 @@ void EditStyle::textStyleValueChanged(TextStylePropertyType type, QVariant value
 
     for (const auto& a : *ts) {
         if (a.type == type) {
-            setStyleQVariantValue(a.sid, value);
+            if (int(type) == int(TextStylePropertyType::MusicalSymbolsScale)) {
```
Comment (cbjeukendrup) [inline src/notation/view/widgets/editstyle.cpp:2629]: Same here (unless I'm mistaken…)

**Review by cbjeukendrup — APPROVED** [review]
Some tiny almost-optional comments but for the rest this looks good to me!

**Pull request merged**